flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -213,19 +215,23 @@ public SingleInputGate(
 	}
 
 	@Override
-	public void setup() throws IOException, InterruptedException {
+	public void setup() throws IOException {
 		checkState(this.bufferPool == null, "Bug in input gate setup logic: Already registered buffer pool.");
 		// assign exclusive buffers to input channels directly and use the rest for floating buffers
 		assignExclusiveSegments();
 
 		BufferPool bufferPool = bufferPoolFactory.get();
 		setBufferPool(bufferPool);
+	}
+
 
-		requestPartitions();
```

Comment:
who is now calling this method in batch tasks?

Reply:
No matter streaming or batch, this method is always called by task thread when the `SingleInputGate` is created to setup. The only difference between streaming and batch is for RPC call `updateInputChannel` which is only valid in batch. But for batch we do not support unaligned cp, so the new transformed remote channel from unknown can request partition directly.